src/components/sections/forcedslots/ExoticSlots.jsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Checkbox,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from '@mui/material';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { makeStyles } from 'tss-react/mui';
+import {
+  changeExotic,
+  changeAllExotic,
+  getExoticsData,
+  getPriority,
+  getExclusionData,
+} from '../../../state/slices/priorities';
+import { GEAR_SLOTS, WeaponTypes } from '../../../utils/gw2-data';
+
+const useStyles = makeStyles()((theme) => ({
+  tableCollapse: {
+    borderCollapse: 'collapse !important',
+    marginBottom: '0px !important',
+  },
+  checkbox: {
+    padding: '6px',
+  },
+}));
+
+const ExoticSlots = () => {
+  const { classes } = useStyles();
+
+  const dispatch = useDispatch();
+  const excludedSlots = useSelector(getExclusionData);
+  const dualWielded = useSelector(getPriority('weaponType'));
+  const affixes = useSelector(getPriority('affixes'));
+  const exotics = useSelector(getExoticsData);
+
+  let SLOTS = GEAR_SLOTS;
+  if (dualWielded !== WeaponTypes.dualWield) {
+    SLOTS = GEAR_SLOTS.slice(0, 13);
+  }
+
+  const handleChange = (index, affix) => (event) => {
+    dispatch(changeExotic({ index, affix, value: event.target.checked }));
+  };
+  const handleChangeAll = (event) => {
+    dispatch(changeAllExotic({ value: event.target.checked }));
+  };
+  const allExotics = Object.values(exotics).flat();
+  const identity = (arg) => arg;
+  const allExoticsChecked = allExotics.length > 0 && allExotics.every(identity);
+  const someExoticsChecked = !allExoticsChecked && allExotics.some(identity);
```

Comment:
It totally works either way, but you can use the `Boolean` builtin for these kinds of things! e.g. `allExotics.every(Boolean)`

Reply:
oh good I was looking for an identity function shorthand